sympy/solvers/ode/tests/test_ode.py

```diff
@@ -32,6 +32,173 @@
+            'sol': [Eq(f(x), C1*x**-3 + C2*x**3)]
+        },
+
+        'euler_hom_07': {
+            'eq': sin(x)*x**2*f(x).diff(x, 2) + sin(x)*x*f(x).diff(x) + sin(x)*f(x),
+            'sol': [Eq(f(x), C1*sin(log(x)) + C2*cos(log(x)))]
+        },
+    }
+}
+
+
+ode_sol_euler_undetermined_coeff = {
+    'hint': "nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients",
+    'examples':{
+        'euler_undet_01': {
+            'eq': Eq(x**2*diff(f(x), x, x) + x*diff(f(x), x), 1),
+            'sol': [Eq(f(x), C1 + C2*log(x) + log(x)**2/2)]
+        },
+
+        'euler_undet_02': {
+            'eq': Eq(x**2*diff(f(x), x, x) - 2*x*diff(f(x), x) + 2*f(x), x**3),
+            'sol': [Eq(f(x), x*(C1 + C2*x + Rational(1, 2)*x**2))]
+        },
+
+        'euler_undet_03': {
+            'eq': Eq(x**2*diff(f(x), x, x) - x*diff(f(x), x) - 3*f(x), log(x)/x),
+            'sol': [Eq(f(x), (C1 + C2*x**4 - log(x)**2/8 - log(x)/16)/x)]
+        },
+
+        'euler_undet_04': {
+            'eq': Eq(x**2*diff(f(x), x, x) + 3*x*diff(f(x), x) - 8*f(x), log(x)**3 - log(x)),
+            'sol': [Eq(f(x), C1/x**4 + C2*x**2 - Rational(1,8)*log(x)**3 - Rational(3,32)*log(x)**2 - Rational(1,64)*log(x) - Rational(7, 256))]
+        },
+
+        'euler_undet_05': {
+            'eq': Eq(x**3*diff(f(x), x, x, x) - 3*x**2*diff(f(x), x, x) + 6*x*diff(f(x), x) - 6*f(x), log(x)),
+            'sol': [Eq(f(x), C1*x + C2*x**2 + C3*x**3 - Rational(1, 6)*log(x) - Rational(11, 36))]
+        },
+    }
+}
+
+
+ode_sol_euler_var_para = {
+    'hint': "nth_linear_euler_eq_nonhomogeneous_variation_of_parameters",
+    'examples':{
+        'euler_var_01': {
+            'eq': Eq(x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x), x**4),
+            'sol': [Eq(f(x), x*(C1 + C2*x + x**3/6))]
+        },
+
+        'euler_var_02': {
+            'eq': Eq(3*x**2*diff(f(x), x, x) + 6*x*diff(f(x), x) - 6*f(x), x**3*exp(x)),
+            'sol': [Eq(f(x), C1/x**2 + C2*x + x*exp(x)/3 - 4*exp(x)/3 + 8*exp(x)/(3*x) - 8*exp(x)/(3*x**2))]
+        },
+
+        'euler_var_03': {
+            'eq': Eq(x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x), x**4*exp(x)),
+            'sol':  [Eq(f(x), x*(C1 + C2*x + x*exp(x) - 2*exp(x)))]
+        },
+
+        'euler_var_04': {
+            'eq': x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x) - log(x),
+            'sol': [Eq(f(x), C1*x + C2*x**2 + log(x)/2 + Rational(3, 4))]
+        },
+
+        'euler_var_05': {
+            'eq': -exp(x) + (x*Derivative(f(x), (x, 2)) + Derivative(f(x), x))/x,
+            'sol': [Eq(f(x), C1 + C2*log(x) + exp(x) - Ei(x))]
+        },
+    }
+}
+
+all_solvers = [ode_sol_euler_homogeneous, ode_sol_euler_undetermined_coeff, ode_sol_euler_var_para]
+all_examples = []
+
+for solver in all_solvers:
+    for example in solver['examples']:
+        all_examples.append(solver['examples'][example]['eq'])
+
 
```

Comment:
@oscarbenjamin does this approach look good for generating all examples in a single structure?